File: src/main/java/com/googongill/aditory/service/dto/category/CreateCategoryResult.java
package com.googongill.aditory.service.dto.category;

import com.googongill.aditory.domain.Category;
import com.googongill.aditory.domain.enums.CategoryState;
import lombok.Getter;
import lombok.Builder;

import java.time.LocalDateTime;
@Getter
@Builder
public class CreateCategoryResult {
    private Long categoryId;
    private String categoryName;
    private CategoryState categoryState;
    private LocalDateTime createdAt;

    public static CreateCategoryResult of(Category category) {
        return CreateCategoryResult.builder()
                .categoryId(category.getId())
                .categoryName(category.getCategoryName())
                .categoryState(category.getCategoryState())
                .createdAt(category.getCreatedAt())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/service/dto/category/CreateCategoryResult.kt
package com.googongill.aditory.service.dto.category

import com.googongill.aditory.domain.Category
import com.googongill.aditory.domain.enums.CategoryState
import java.time.LocalDateTime

data class CreateCategoryResult(
    val categoryId: Long?,
    val categoryName: String,
    val categoryState: CategoryState,
    val createdAt: LocalDateTime?,
) {

    companion object {
        fun of(category: Category): CreateCategoryResult {
            return CreateCategoryResult(
                categoryId = category.id,
                categoryName = category.categoryName,
                categoryState = category.categoryState,
                createdAt = category.createdAt
            )
        }
    }
}
